cloud/amazon/ec2_vpc_nat_gateway.py

```diff
@@ -0,0 +1,298 @@
+            if nat_gateway_status_list[0].get('state') in ('deleted', 'absent'):
+                module.exit_json(**results)
+            else:
+                time.sleep(5)
+        module.fail_json(msg="Waited too long for VPC NAT Gateway to be deleted.")
+    else:
+        module.exit_json(**results)
+
+
+def create_nat_gateway(ec2_client=None, module=None):
+    """ Create the NAT gateway """
+    try:
+        result = ec2_client.create_nat_gateway(SubnetId=module.params.get('subnet_id'),
+                                               AllocationId=module.params.get('allocation_id'))
+        wait = module.params.get('wait')
+        start_time = datetime.datetime.utcnow()
+        if wait:
+            wait_timeout = time.time() + module.params.get('wait_timeout')
+            while wait_timeout > time.time():
+                present_status_list = get_nat_gateway_status_list(ec2_client=ec2_client, module=module)
+                for present_status in present_status_list:
+                    create_time = present_status.get('created')
+                    if present_status.get('state') == 'failed' and create_time.replace(tzinfo=None) >= start_time:
+                        module.fail_json(msg="Failed to create VPC NAT Gateway")
+                    elif present_status.get('state') == 'available':
+                        nat_gateway_id = result['NatGateway']['NatGatewayId']
+                        results = dict(changed=True, nat_gateway_id=nat_gateway_id)
+                        module.exit_json(**results)
+                else:
+                    time.sleep(5)
+            module.fail_json(msg="Waited too long for VPC NAT Gateway to be created.")
+        else:
+            module.exit_json(changed=True)
+    except botocore.exceptions.ClientError as ce:
+        module.fail_json(msg=ce.message)
+
+
+def ensure_nat_gateway_absent(ec2_client=None, module=None):
+    """ Ensure the specified NAT gateway does not exist and call delete if it does """
+    try:
+        results = dict(changed=False)
+        nat_gateways_result = ec2_client.describe_nat_gateways(NatGatewayIds=[module.params.get('nat_gateway_id')])
+        if nat_gateways_result:
+            nat_gateway = nat_gateways_result.get('NatGateways')[0]
+            if nat_gateway.get('State') in ('pending',
+                                            'available'):
+                if module.check_mode:
+                    results['changed'] = True
+                    module.exit_json(**results)
+                else:
+                    delete_nat_gateway(ec2_client=ec2_client,
+                                       module=module,
+                                       nat_gateway=nat_gateway)
+        module.exit_json(**results)
+    except botocore.exceptions.ClientError as ce:
+        if "NatGatewayMalformed" in ce.message:
+            module.fail_json(msg="Invalid NAT Gateway ID")
```

Comment:
Don't look for a string in message.  Use error code provided instead.

E.g...
if e.response['Error']['Code'] == 'bla':